nano/node/active_transactions.cpp

```diff
@@ -579,6 +588,80 @@ void nano::active_transactions::erase (nano::block const & block_a)
 	}
 }
 
+bool nano::active_transactions::should_flush ()
+{
+	bool result (false);
+	counter.trend_sample ();
+	size_t minimum_size (1);
+	std::unique_lock<std::mutex> counter_lock (counter.mutex);
+	if (counter.rate == 0)
+	{
+		minimum_size = 4;
+	}
+	else
+	{
+		minimum_size = counter.rate * 10;
+	}
+	if (roots.size () > minimum_size)
+	{
+		if (counter.rate <= 10)
+		{
+			if (static_cast<double> (roots.size () * .75) < long_unconfirmed_size ())
+			{
+				result = true;
+			}
+		}
+		else if (counter.rate <= 100)
+		{
+			if (static_cast<double> (roots.size () * .50) < long_unconfirmed_size ())
+			{
+				result = true;
+			}
+		}
+		else if (counter.rate <= 1000)
+		{
+			if (static_cast<double> (roots.size () * .25) < long_unconfirmed_size ())
+			{
+				result = true;
+			}
+		}
+	}
+	counter_lock.unlock ();
+	return result;
+}
+
+uint64_t nano::active_transactions::flush_lowest ()
+{
+	uint64_t result (0);
+	size_t count (0);
+	assert (!roots.empty ());
+	auto & sorted_roots = roots.get<1> ();
+	for (auto it = sorted_roots.rbegin (); it != sorted_roots.rend ();)
+	{
+		if (count != 2)
+		{
+			auto election = it->election;
+			if (election->announcements > announcement_long && !election->confirmed && !node.wallets.watcher.is_watched (it->root))
+			{
+				result = it->difficulty;
```

Comment:
Should it be adjusted difficulty?

Reply:
this is an artifact from some debug testing I was doing, it doesn't actually matter one way or another, will probably void this function, there is nothing needing to be returned